pkg/cmd/server/kubernetes/master.go

```diff
@@ -58,39 +62,39 @@ func (c *MasterConfig) RunNamespaceController() {
 	namespaceController := namespacecontroller.NewNamespaceController(c.KubeClient, experimentalMode, c.ControllerManager.NamespaceSyncPeriod)
 	namespaceController.Run()
 }
-
-// RunPersistentVolumeClaimBinder starts the Kubernetes Persistent Volume Claim Binder
-func (c *MasterConfig) RunPersistentVolumeClaimBinder() {
-	binder := volumeclaimbinder.NewPersistentVolumeClaimBinder(c.KubeClient, c.ControllerManager.PVClaimBinderSyncPeriod)
-	binder.Run()
-}
-
 func (c *MasterConfig) RunPersistentVolumeClaimRecycler(recyclerImageName string) {
 	defaultScrubPod := volume.NewPersistentVolumeRecyclerPodTemplate()
 	defaultScrubPod.Spec.Containers[0].Image = recyclerImageName
 	defaultScrubPod.Spec.Containers[0].Command = []string{"/usr/share/openshift/scripts/volumes/recycler.sh"}
 	defaultScrubPod.Spec.Containers[0].Args = []string{"/scrub"}
 
+	volumeConfig := c.ControllerManager.VolumeConfigFlags
 	hostPathConfig := volume.VolumeConfig{
-		RecyclerMinimumTimeout:   30,
-		RecyclerTimeoutIncrement: 30,
+		RecyclerMinimumTimeout:   volumeConfig.PersistentVolumeRecyclerMinimumTimeoutHostPath,
+		RecyclerTimeoutIncrement: volumeConfig.PersistentVolumeRecyclerIncrementTimeoutHostPath,
 		RecyclerPodTemplate:      defaultScrubPod,
 	}
 	nfsConfig := volume.VolumeConfig{
-		RecyclerMinimumTimeout:   180,
-		RecyclerTimeoutIncrement: 30,
+		RecyclerMinimumTimeout:   volumeConfig.PersistentVolumeRecyclerMinimumTimeoutNFS,
+		RecyclerTimeoutIncrement: volumeConfig.PersistentVolumeRecyclerIncrementTimeoutNFS,
 		RecyclerPodTemplate:      defaultScrubPod,
 	}
 
 	allPlugins := []volume.VolumePlugin{}
 	allPlugins = append(allPlugins, host_path.ProbeVolumePlugins(hostPathConfig)...)
 	allPlugins = append(allPlugins, nfs.ProbeVolumePlugins(nfsConfig)...)
-
-	recycler, err := volumeclaimbinder.NewPersistentVolumeRecycler(c.KubeClient, c.ControllerManager.PVClaimBinderSyncPeriod, allPlugins)
+	allPlugins = append(allPlugins, aws_ebs.ProbeVolumePlugins()...)
+	allPlugins = append(allPlugins, gce_pd.ProbeVolumePlugins()...)
+	allPlugins = append(allPlugins, cinder.ProbeVolumePlugins()...)
+
+	qosClasses := c.ControllerManager.VolumeConfigFlags.ProvisionerQualityOfServiceClasses
```

Comment:
Given the risk involved in other changes, I don't want to land this until
it's had bake time.  I don't want that bake time to be in Origin.

On Thu, Oct 22, 2015 at 2:25 PM, Mark Turansky <notifications@github.com>
wrote:

> In pkg/cmd/server/kubernetes/master.go
> <https://github.com/openshift/origin/pull/5217#discussion_r42786220>:
>
> >  		RecyclerPodTemplate:      defaultScrubPod,
> >  	}
> >
> >  	allPlugins := []volume.VolumePlugin{}
> >  	allPlugins = append(allPlugins, host_path.ProbeVolumePlugins(hostPathConfig)...)
> >  	allPlugins = append(allPlugins, nfs.ProbeVolumePlugins(nfsConfig)...)
> > -
> > -	recycler, err := volumeclaimbinder.NewPersistentVolumeRecycler(c.KubeClient, c.ControllerManager.PVClaimBinderSyncPeriod, allPlugins)
> > +	allPlugins = append(allPlugins, aws_ebs.ProbeVolumePlugins()...)
> > +	allPlugins = append(allPlugins, gce_pd.ProbeVolumePlugins()...)
> > +	allPlugins = append(allPlugins, cinder.ProbeVolumePlugins()...)
> > +
> > +	qosClasses := c.ControllerManager.VolumeConfigFlags.ProvisionerQualityOfServiceClasses
>
> We've tested through all of it and it also fixes a race bug.
>
> I don't expect a lot of upstream change, but I also cannot accurately
> guess how much they will want. It might be 0.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/5217/files#r42786220>.
>